### PR TITLE
[24862] makes User mpi and mpi_profile methods private

### PR DESCRIPTION
**Pull request by bramleyjl**
## Description of change

This change removes the last few calls of the User model's `mpi` method and makes it private. It also returns the `mpi_profile` method to private as well and cleans up some testing surrounding these two changes.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#24862

## Things to know about this PR

Affected unit specs update and manual login/logout workflow.

### CODE REVIEW

**Review by cbtkc — DISMISSED** [review]
Looks good

**Comment by LindseySaari** [comment]
Looks like some linting issues. Can take another 👀  once that's resolved

**Comment by patrickvinograd** [comment]
Please hold on merging this until the mobile team has a chance to review. I will follow up on it this morning.

**Comment by patrickvinograd** [comment]
I think we're all clear from mobile side, though I'd advocate for merging this after today's deploy so we have some time to observe it in staging.

**Review by ericboehs — APPROVED** [review]

File: app/controllers/v0/profile/personal_informations_controller.rb [hunk 1]
```diff
@@ -7,22 +7,13 @@ class PersonalInformationsController < ApplicationController
...
-      #       }
-      #     }
-      #   }
-      #
       def show
-        response = @current_user.mpi.profile
-
+        response = OpenStruct.new({
+                                    'id': @current_user.account_uuid,
+                                    'type': 'mvi_models_mvi_profiles',
```
Comment (bosawt) [inline app/controllers/v0/profile/personal_informations_controller.rb:13]: Setting `type` here doesn't do anything, it gets overridden when serialized (it'll be `open_structs` to match the type of the class that we serialize). There are ways we can fix it, but I don't think the frontend cares what the `type` is here, so I think we should just remove this line and let the type be `open_structs`)

**Review by bosawt — APPROVED** [review]
I logged in a user and made a curl to the `personal_inormations` route, everything looks good! 

I have a couple quick comments that are really just cleaning up the code a bit.

**Pull request merged**